app/models/concerns/course/searchable_concern.rb

```diff
@@ -0,0 +1,18 @@
+module Course::SearchableConcern
+  extend ActiveSupport::Concern
+
+  module ClassMethods
```

Comment:
I think this should not be generic, the fields that a model wants to be searched by will differ.